src/modules/75-cd/components/EnvironmentGroups/CreateEnvironmentGroupModal.tsx

```diff
@@ -0,0 +1,418 @@
+                      <Button
+                        variation={ButtonVariation.TERTIARY}
+                        text={getString('cancel')}
+                        onClick={closeModal}
+                        disabled={createLoading}
+                      />
+                    </Layout.Horizontal>
+                  </Layout.Vertical>
+                  <Layout.Vertical padding={{ left: 'large' }} flex={{ alignItems: 'stretch' }} width={'50%'}>
+                    <ExpandingSearchInput
+                      alwaysExpanded
+                      placeholder={'Search Environments'}
+                      autoFocus={false}
+                      width={'100%'}
+                      onChange={setSearchTerm}
+                      throttle={200}
+                    />
+                    <Layout.Vertical
+                      height={300}
+                      style={{
+                        overflow: 'scroll'
+                      }}
+                    >
+                      {isFetchingEnvironmentsFirstTime ? (
+                        <Container flex={{ align: 'center-center' }} height={'100%'}>
+                          <Spinner size={32} />
+                        </Container>
+                      ) : isEmptyContent ? (
+                        <Layout.Vertical flex={{ align: 'center-center' }} width={'100%'} height={'100%'}>
+                          <img src={EmptyEnvironmentGroup} alt={getString('cd.noEnvironment.title')} />
+                          <Heading level={2}>{getString('cd.noEnvironment.title')}</Heading>
+                        </Layout.Vertical>
+                      ) : (
+                        (environments as EnvironmentResponse[])?.map((item, index) => {
+                          if (!item?.environment) {
+                            return null
+                          }
+
+                          const { name, identifier, tags } = item?.environment
+                          const checked = selectedEnvironments.some((_id: string) => _id === identifier)
+
+                          return (
+                            <Layout.Horizontal
+                              width={'100%'}
+                              height={60}
+                              flex={{ justifyContent: 'flex-start', alignItems: 'center' }}
+                              padding={{ left: 'medium' }}
+                              margin={{ top: 0 }}
+                              border={{ bottom: true }}
+                              key={identifier}
+                              ref={attachRefToLastElement(index) ? loadMoreRef : undefined}
+                            >
+                              <EnvironmentSelection
+                                identifier={identifier}
+                                checked={checked}
+                                onChange={(e: FormEvent<HTMLInputElement>) => {
+                                  if ((e.target as any).checked && identifier) {
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232281)
```suggestion
                                  if ((e.target as unknown).checked && identifier) {
```